src/NameService/NameService.cs

```diff
@@ -474,12 +474,13 @@ private static bool CheckIPv6(string ipv6)
             length = fragments.Length;
             if (length < 3 || length > 8) return false;
             ushort[] numbers = new ushort[8];
-            bool hasEmpty = false;
+            bool hasEmpty = false, isCompressed = false;
             for (int i = 0; i < length; i++)
             {
                 string fragment = fragments[i];
                 if (fragment.Length == 0)
                 {
+                    isCompressed = true;
```

Comment:
IIUC this will allow for `aabb:` address which in fact is invalid (while `aabb::` is).

Reply:
Please check #7